### PR TITLE
Core/Garrisons Fixes

### PR DESCRIPTION
**Pull request by soulfrost**

### CODE REVIEW

File: src/server/game/Garrison/Garrison.cpp [hunk 1]
```diff
@@ -483,11 +478,7 @@ void Garrison::AddMission(uint32 garrMissionId)
     mission.PacketInfo.MissionRecID = garrMissionId;
     mission.PacketInfo.OfferTime = time(nullptr);
     mission.PacketInfo.OfferDuration = missionEntry->OfferTime;
-    mission.PacketInfo.StartTime = time_t(2254525440);
-    mission.PacketInfo.TravelDuration = missionEntry->TravelTime;
-    mission.PacketInfo.MissionDuration = missionEntry->Duration;
```
Comment (Traesh) [inline src/server/game/Garrison/Garrison.cpp:488]: Why are you removing those 3 lines ?

File: src/server/game/Garrison/Garrison.cpp [hunk 2]
```diff
@@ -589,8 +580,8 @@ std::pair<std::vector<GarrMissionEntry const*>, std::vector<double>> Garrison::G
         if (missionEntry->RequiredFollowersCount > activeFolowerCount)
             continue;
 
-        if (missionEntry->RequiredItemLevel > averageFollowerILevel)
-            continue;
+        //if (missionEntry->RequiredItemLevel > averageFollowerILevel)
+        //    continue;
```
Comment (Traesh) [inline src/server/game/Garrison/Garrison.cpp:584]: Why commenting ?

Comment (soulfrost) [reply src/server/game/Garrison/Garrison.cpp:584]: this's wrong, example: mission required 750 follower Item level for maching bias, but ur followers has just 730 ilvl - u still'll be able to send forllowers on this mission but with lower succes chanse

Comment (soulfrost) [reply src/server/game/Garrison/Garrison.cpp:584]:
```c#

		public static float ComputeFollowerBias(WrapperGarrisonFollower follower, int followerLevel, int followerItemLevel, int garrMissionID)
		{
			bool flag = (follower.Flags & 8) != 0;
			if (flag)
			{
				return 0f;
			}
			GarrMissionRec record = StaticDB.garrMissionDB.GetRecord(garrMissionID);
			if (record == null)
			{
				return 0f;
			}
			float num = MissionDetailView.ComputeFollowerLevelBias(follower, followerLevel, record.TargetLevel);
			num += MissionDetailView.ComputeFollowerItemLevelBias(follower, followerItemLevel, (int)record.GarrFollowerTypeID, (int)record.TargetItemLevel);
			return Mathf.Clamp(num, -1f, 1f);
		}

		private static float ComputeFollowerItemLevelBias(WrapperGarrisonFollower follower, int followerItemLevel, int garrFollowerTypeID, int targetItemLevel)
		{
			float num = 0f;
			if (follower.FollowerLevel == MissionDetailView.GarrisonFollower_GetMaxFollowerLevel(garrFollowerTypeID))
			{
				int num2;
				int num3;
				MissionDetailView.GarrisonFollower_GetFollowerBiasConstants(follower, out num2, out num3);
				int num4 = (targetItemLevel <= 0) ? 600 : targetItemLevel;
				float num5 = (float)(followerItemLevel - num4);
				num5 /= (float)num3;
				num += num5;
			}
			return num;
		}

		private static float ComputeFollowerLevelBias(WrapperGarrisonFollower follower, int followerLevel, int targetLevel)
		{
			int num;
			int num2;
			MissionDetailView.GarrisonFollower_GetFollowerBiasConstants(follower, out num, out num2);
			float num3 = (float)(followerLevel - targetLevel);
			return num3 / (float)num;
		}

```

Comment (soulfrost) [reply src/server/game/Garrison/Garrison.cpp:584]: partyal example how it calcs

File: src/server/game/Garrison/Garrison.cpp (hunk 1 above)
Comment (Traesh) [reply src/server/game/Garrison/Garrison.cpp:488]: Confirmed for unstarted missions, but are your sure about it even when mission is already started (MissionState != 0) ?

And then you need to assign it in Garrison::StartMission or they'll always be default values

File: src/server/game/Garrison/Garrison.cpp (hunk 2 above)
Comment (Traesh) [reply src/server/game/Garrison/Garrison.cpp:584]: Ok thanks for the infos

File: src/server/game/Garrison/Garrison.cpp (hunk 1 above)
Comment (soulfrost) [reply src/server/game/Garrison/Garrison.cpp:488]: cant find any sniffs where sended this packet with State != 0

**Pull request merged**